bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ProjectRepository.cs

```diff
@@ -60,16 +60,23 @@ public async Task<IEnumerable<ProjectPermissionDetails>> GetManyByOrganizationId
 
     public async Task DeleteManyByIdAsync(IEnumerable<Guid> ids)
     {
-        using (var scope = ServiceScopeFactory.CreateScope())
+        using var scope = ServiceScopeFactory.CreateScope();
+        var utcNow = DateTime.UtcNow;
+        var dbContext = GetDatabaseContext(scope);
+        var projects = dbContext.Project
+            .Where(c => ids.Contains(c.Id))
+            .Include(p => p.Secrets);
+        await projects.ForEachAsync(project =>
         {
-            var dbContext = GetDatabaseContext(scope);
-            var projects = dbContext.Project.Where(c => ids.Contains(c.Id));
-            await projects.ForEachAsync(project =>
+            foreach (var projectSecret in project.Secrets)
             {
-                dbContext.Remove(project);
-            });
-            await dbContext.SaveChangesAsync();
-        }
+                projectSecret.RevisionDate = utcNow;
+            }
+
+            dbContext.Remove(project);
```

Comment:
🌱 I wish we had https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#executeupdate-and-executedelete-bulk-updates for this ...